kive/container/test_dataset_comparison.py

```diff
@@ -0,0 +1,596 @@
+import datetime
+import io
+import random
+import typing as ty
+from unittest import mock
+
+from django.core.files import File
+from django.test import TestCase
+
+from librarian.models import Dataset
+import metadata.models
+
+from .models import (
+    Container,
+    ContainerApp,
+    ContainerDataset,
+    ContainerFamily,
+    ContainerArgument,
+    ContainerArgumentType,
+    ContainerRun,
+)
+from . import runutils
+
+
+def random_string() -> str:
+    from string import ascii_letters
+    return ''.join(random.sample(ascii_letters, 32))
+
+
+def fake_file(name: str, content: str) -> ty.IO:
```

Comment:
Will [`ContentFile`] do what you need here?

[`ContentFile`]: https://docs.djangoproject.com/en/3.0/ref/files/file/#the-contentfile-class

Reply:
Ooh, yes it will. I even picked the same argument names. 😆